test/test_respondent_controller.py

```diff
@@ -2032,3 +2035,84 @@ def _enroll_respondent(self, session):
         session.flush()
         session.refresh(respondent)
         return respondent.to_respondent_dict()
+
+    @patch("ras_party.controllers.respondent_controller.send_account_deletion_confirmation_email")
+    @patch("ras_party.controllers.respondent_controller.session", new_callable=MagicMock)
+    def test_all_respondents_marked_are_delete(self, mock_session, mock_send_account_deletion_confirmation_email):
+        respondent_1 = Respondent()
+        respondent_1.email_address = "test1@example.com"
+        respondent_1.first_name = "One"
+
+        respondent_2 = Respondent()
+        respondent_2.email_address = "test2@example.com"
+        respondent_2.first_name = "Two"
+
+        respondent_3 = Respondent()
+        respondent_3.email_address = "test3@example.com"
+        respondent_3.first_name = "Three"
+
+        respondents_to_delete = [respondent_1, respondent_2, respondent_3]
+
+        mock_query = MagicMock()
+        mock_query.filter(Respondent.mark_for_deletion).return_value = respondents_to_delete
+        # To iterate over "for respondent in respondents" the __iter__ method needs defining for the Mock
+        mock_query.filter.return_value.__iter__.return_value = iter(respondents_to_delete)
+        mock_session.query.return_value = mock_query
+
+        # Execute
+        with self.app.app_context():
+            delete_respondents_marked_for_deletion.__wrapped__(mock_session)
+
+        # Verify
+        expected_confirmation_email_calls = [
+            call(respondent_1.email_address, respondent_1.first_name),
+            call(respondent_2.email_address, respondent_2.first_name),
+            call(respondent_3.email_address, respondent_3.first_name),
+        ]
+        mock_send_account_deletion_confirmation_email.assert_has_calls(expected_confirmation_email_calls)
+        self.assertEqual(mock_session.commit.call_count, 3)
+        mock_session.rollback.assert_not_called()
+
+    @patch("ras_party.controllers.respondent_controller.send_account_deletion_confirmation_email")
+    @patch("ras_party.controllers.respondent_controller.session", new_callable=MagicMock)
+    def test_delete_respondents_continues_when_exception(
+        self, mock_session, mock_send_account_deletion_confirmation_email
+    ):
+        # Setup
+        respondent_1 = Respondent()
+        respondent_1.email_address = "test1@example.com"
+        respondent_1.first_name = "One"
+
+        respondent_2 = Respondent()
+        respondent_2.email_address = "test2@example.com"
+        respondent_2.first_name = "Two"
+
+        respondent_3 = Respondent()
+        respondent_3.email_address = "test3@example.com"
+        respondent_3.first_name = "Three"
+
+        respondents_to_delete = [respondent_1, respondent_2, respondent_3]
+
+        # Test currently failing, I need to work out how to throw an exception ONLY for respondent_2
+        mock_session.side_effect = IntegrityError("Integrity error", "params", "orig")
```

Comment:
This is correctly failing at the moment. I'm trying to work out how to throw an exception _only_ for respondent_2 to replicate the production bug.